### PR TITLE
[7.4-only] LPS-121853 Renaming in management toolbar to match naming convention

### PR DESCRIPTION
**Pull request by markocikos**
This is a followup on https://github.com/liferay-frontend/liferay-portal/pull/639, a rename based on https://github.com/brianchandotcom/liferay-portal/pull/97852#issuecomment-762859400

This seemingly innocent rename is problematic, we already discussed it in https://github.com/liferay-frontend/liferay-portal/pull/639#discussion_r555943271

The problem is that `ManagementToolbarDisplayContext getDisplayContext()` does not fit naming convention.

In this PR, we are fixing the problem the cleanest way, that keeps model class naming consistent with other tags:
- We are renaming model object to `ManagementToolbar`, while still keeping the `ManagementToolbarDisplayContext` object that is used directly in several places. Even if they are (currently) the same, it makes sense to me to keep these as separate classes. They represent different concepts, where model of a component is a subset of display context.
- We are renaming attribute to `managementToolbar`, while deprecating `displayContext`

The downside of the solution in this PR is we have this bit of awkwardness:
```jsp
<clay:management-toolbar
	managementToolbar="<%= selectAccountEntryAddressManagementToolbarDisplayContext %>"
/> 
```
Is this acceptable? If not, we would need to replace every `*ManagementToolbarDisplayContext.java` class into `*ManagementToolbar.java`. We can do it during migration, but it is worth the effort?

There are a few alternative solutions I considered:
- Rename `ManagementToolbarDisplayContext` to `DisplayContext`
  -  `DisplayContext` is very generic, while everything in that class is related to the management toolbar component. It is also used directly in several places, so there would be some refactoring.
- Rename `displayContext` attribute to `managementToolbarDisplayContext`. 
  - Besides a bunch of refactors in the tag class, main gripe I have is an ugly-looking needless prefix for an attribute in the `management-toolbar` tag.
  - Here is [my branch with this solution](https://github.com/markocikos/liferay-portal/commits/LPS-121853-2-alternative-2), I can quickly force-push if we prefer it.
- Move content of `ManagementToolbarDisplayContext` class to  `.management.toolbar.DisplayContext`, keep `displayContext` attribute
  - This solution would result in the least amount of present and future refactoring. But, we would need to add a bit awkward new package, and we would have a public class that is poorly descriptive of what it does. You would need to pay attention to the package name.
  - Here is [my branch with this solution](https://github.com/markocikos/liferay-portal/commits/LPS-121853-2-alternative), I can quickly force-push if we prefer it.

None of the alternatives offer consistent model class naming with other tags. 

@carloslancha @jbalsas @julien @brianchandotcom

### CODE REVIEW

**Comment by jbalsas** [comment]
Changes look good to me, kinda make sense, are more in sync with other tags and tests are green so I don't think anything major is broken... this is a go from me

sealofapproval

If no one objects or provides additional feedback I'll forward it later today.

**Review by jbalsas — APPROVED** [review]

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/management_toolbars.jsp [hunk 1]
```diff
@@ -68,6 +68,6 @@ ClaySampleManagementToolbarsDisplayContext managementToolbarsDisplayContext = ne
 <h3>USING DISPLAY CONTEXT</h3>
 
 <clay:management-toolbar
-	displayContext="<%= managementToolbarsDisplayContext %>"
+	managementToolbar="<%= managementToolbarsDisplayContext %>"
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/management_toolbars.jsp:71]: It's a bit weird that this is `managementToolbar**s**DisplayContext` in plural 😂

**Review by carloslancha — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia4.giphy.com%2Fmedia%2FPudZiAbQDUEik%2Fgiphy-downsized-medium.gif&token=approve" data-gitmeme-token="approve"></a>

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/management_toolbars.jsp (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/partials/management_toolbars.jsp:71]: There are more toolbars on the same page in `sample-web`, and the page is named `Management Toolbars`. Makes sense, just barely :slightly_smiling_face:

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**